dbcon/mysql/ha_mcs_impl.cpp

```diff
@@ -3669,8 +3677,8 @@ int ha_mcs_impl_end_bulk_insert(bool abort, TABLE* table)
              ((thd->lex)->sql_command == SQLCOM_INSERT) || ((thd->lex)->sql_command == SQLCOM_LOAD) ||
              ((thd->lex)->sql_command == SQLCOM_INSERT_SELECT) || ci->isCacheInsert) )
         {
+            bitmap_clear_all(table->read_set);
```

Comment:
I find it too annoying to add bitmap_clear_all() in so many places.

Please consider adding a class to clear the bitmap automatically when the execution leaves the block.
```
class Scoped_bitmap_set_and_clear_all
{
  MY_BITMAP *mBitmap;
public:
  Scoped_bitmap_clear_all(MY_BITMAP *bitmap)
     :Bitmap(bitmap)
  {
    bitmap_set_all(mBitmap);
  }
  ~Scoped_bitmap_clear_all()
  {
    bitmap_clear_all(mBitmap);
  }
};
```

Reply:
I don't see how it would fit in here w/o adding another member into ci variable that must be properly cleared in many places.